packages/arb-validator/rollupvalidator/txtracker.go

```diff
@@ -110,214 +136,290 @@ func (a *assertionInfo) FindLogs(address *big.Int, topics []common.Hash) []logRe
 				}
 			}
 			if match {
-				logs = append(logs, logResponse{evmLog, txLogs.msg})
+				logs = append(logs, logResponse{
+					Log:     evmLog,
+					TxIndex: txLogs.TxIndex,
+					TxHash:  txLogs.TxHash,
+				})
 			}
 		}
 	}
 	return logs
 }
 
-func newAssertionInfo() *assertionInfo {
-	return &assertionInfo{}
-}
-
 type txTracker struct {
-	txRequestIndex int
-	transactions   map[common.Hash]txInfo
-	assertionInfo  []*assertionInfo
-	assertionMap   map[common.Hash]*assertionInfo
-	accountNonces  map[common.Address]uint64
-	vmID           common.Address
-	requests       chan validatorRequest
+	sync.RWMutex
+	rollup.NoopListener
```

Comment:
Why use a NoopListener?  It looks like it doesn't do anything.

Reply:
It's because the Listener interface has a large number of different methods that must be implemented, but this class only cares about a few of them. This way we don't need to include noop implementations of all the other interface requirements.